packages/camera/camera_windows/windows/test/mocks.h

```diff
@@ -206,12 +208,14 @@ class MockCamera : public Camera {
   MOCK_METHOD(bool, InitCamera,
               (flutter::TextureRegistrar * texture_registrar,
                flutter::BinaryMessenger* messenger,
-               ResolutionPreset resolution_preset,
-               const RecordSettings& record_settings),
+               const PlatformMediaSettings& media_settings),
               (override));
 
   std::unique_ptr<CaptureController> capture_controller_;
-  std::unique_ptr<MethodResult<>> pending_result_;
+  std::function<void(std::optional<FlutterError> reply)> pending_void_result_;
+  std::function<void(ErrorOr<int64_t> reply)> pending_int_result_;
+  std::function<void(ErrorOr<std::string> reply)> pending_string_result_;
+  std::function<void(ErrorOr<PlatformSize> reply)> pending_size_result_;
```

Comment:
Since we only ever use one pending result at a time in tests, and it's test-only code where changes are less risky, I split these out to preserve full type safety, as this makes tests much easier to write as compared to doing the `variant` thing here.